x-pack/legacy/plugins/canvas/public/plugin.tsx

```diff
@@ -66,21 +63,42 @@ export class CanvasPlugin
     // Things like registering functions to the interpreter that need
     // to be available everywhere, not just in Canvas
 
+    core.application.register({
+      id: 'canvas',
+      title: 'Canvas App',
+      async mount(context, params) {
+        // Load application bundle
+        const { renderApp } = await import('./applications/canvas');
+
+        // Setup our store
+        const canvasStore = await createStore(core);
+
+        // Get start services
+        const [coreStart, depsStart] = await core.getStartServices();
+
+        return renderApp(coreStart, depsStart, params, canvasStore);
+      },
+    });
     return {};
   }
 
   public start(core: CoreStart, plugins: CanvasStartDeps) {
     loadExpressionTypes();
     loadTransitions();
 
-    initStateManagement(core, plugins);
-    initLocationProvider(core, plugins);
-    initStore(core, plugins);
     initClipboard(plugins.__LEGACY.storage);
     initLoadingIndicator(core.http.addLoadingCountSource);
 
-    const CanvasRootController = CanvasRootControllerFactory(core, plugins);
-    plugins.__LEGACY.setRootController('canvas', CanvasRootController);
+    core.chrome.setBadge(
+      core.application.capabilities.canvas && core.application.capabilities.canvas.save
```

Comment:
Originally this was checking for the existence just of `core.application.capabilities.canvas.save` but I was running into a test error where `.canvas` wasn't defined on `capabilities`. I fixed it by adding this check. If we want to fix this by doing a mock somewhere for the test instead, I can do that